packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/metadata.py

```diff
@@ -86,6 +88,18 @@ def initialize(
         )
 
 
+def validate_display_name(display_name: str) -> None:
+    """Validate and raise exception if 'display_name' has invalid characters.
+
+    Raises:
+        ValueError: If display_name has invalid characters.
+    """
+    if display_name == "" or not display_name[0].isalpha():
+        raise ValueError("First letter in " + display_name + " must be a letter")
+    elif not all(char in _VALID_CHARS or char.isalnum() for char in display_name):
+        raise ValueError("Invalid characters in " + display_name)
```

Comment:
Several things here:
- Use f-strings.
- Feel free to use sentences and punctuation.
- Put quotes around the user data so it doesn't blend in with the error text. This is particularly important for empty strings, otherwise you end up with a double space or trailing space in a message that does not make sense.
- Consider making empty string a separate case anyway, since it's not like an empty string has a first letter. Suggestion: "The display name must not be an empty string."
- Use the words "display name" because it may not be obvious from the data. Example: "First letter in 7 must be a letter". Suggestion: "First letter in display name '7' must be a letter."
- "First letter in x must be a letter" is a tautology. Suggestion: "First character in x must be a letter".